app/controllers/request_controller.rb

```diff
@@ -167,6 +167,12 @@ def similar
   end
 
   def list
+    # respond with a 404 and do not execute the search if request was not for html
+    if request.format && !request.format.html?
+      respond_to { |format| format.any { head :not_found } }
+      return
+    end
```

Comment:
The comment talking about "search" is probably wrong though - lazy copy and paste from 67740f0 :(

Agree that intent is clearer with `:respond_to`